cli/template/base/README.md

```diff
@@ -60,55 +66,79 @@ You can also dockerize this stack and deploy a container.
 
   </details>
 
-3. Create a `Dockerfile` with the following contents:
+4. Create a `Dockerfile` with the following contents:
    <details>
    <summary>Dockerfile</summary>
 
    ```Dockerfile
+   ########################
+   #         DEPS         #
+   ########################
+
    # Install dependencies only when needed
-   FROM node:16-alpine AS deps
-   # Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
+   # TODO re-evaluate if emulation is still necessary on arm64 after moving to    node 18
+   FROM --platform=linux/amd64 node:16-alpine AS deps
+   # Check https://github.com/nodejs/docker-node/tree/   b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why    libc6-compat might be needed.
    RUN apk add --no-cache libc6-compat
    WORKDIR /app
 
    # Install dependencies based on the preferred package manager
    COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
    RUN \
-      if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
-      elif [ -f package-lock.json ]; then npm ci; \
-      elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i; \
-      else echo "Lockfile not found." && exit 1; \
-      fi
+     if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
+     elif [ -f package-lock.json ]; then npm ci; \
+     elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm i; \
+     else echo "Lockfile not found." && exit 1; \
+     fi
 
+   ########################
+   #        BUILDER       #
+   ########################
 
    # Rebuild the source code only when needed
-   FROM node:16-alpine AS builder
+   # TODO re-evaluate if emulation is still necessary on arm64 after moving to    node 18
+   FROM --platform=linux/amd64 node:16-alpine AS builder
+
+   ARG NEXT_PUBLIC_FOO
+   ARG BAR
+
    WORKDIR /app
    COPY --from=deps /app/node_modules ./node_modules
    COPY . .
 
    # Next.js collects completely anonymous telemetry data about general usage.
    # Learn more here: https://nextjs.org/telemetry
-   # Uncomment the following line in case you want to disable telemetry during the build.
+   # Uncomment the following line in case you want to disable telemetry during the    build.
    # ENV NEXT_TELEMETRY_DISABLED 1
 
-   RUN yarn build
+   RUN \
+     if [ -f yarn.lock ]; then yarn build; \
+     elif [ -f package-lock.json ]; then npm run build; \
+     elif [ -f pnpm-lock.yaml ]; then yarn global add pnpm && pnpm run build; \
```

Comment:
Looks a bit redundant since pnpm is already installed when installing the packages

Why do we even need to run build with our used package manager? Won't `yarn build` just do the job like before?

Reply:
Does seem a bit misleading that we install pnpm when using yarn but not with npm, should probably stick to the one they are using, i.e.
```Dockerfile
   RUN \
     if [ -f yarn.lock ]; then yarn build; \
     elif [ -f package-lock.json ]; then npm run build; \
     elif [ -f pnpm-lock.yaml ]; then pnpm build; \

Reply:
Here's the PR where this was added in the Next.js with-docker example: https://github.com/vercel/next.js/pull/38693
(and here's the PR where they switched to if/elif https://github.com/vercel/next.js/pull/39072 )

The reasoning makes sense to me. What do you think?